docs/tutorial/durham_reframe_tutorial.md

```diff
@@ -0,0 +1,409 @@
+Before proceeding to install ReFrame, we recommend creating a python virtual environment to avoid clashes with other installed python packages. First load a newer python module.
+
+```bash
+module swap python/3.10.12
+```
+
+Then create an environment and activate it with
+
+```bash
+python3 -m venv reframe_tutorial
+source reframe_tutorial/bin/activate
+```
+
+You will have to activate the environment each time you login. To deactivate the environment run `deactivate`.
+
+----
+
+## Install ReFrame
+
+Then install ReFrame with `pip`. I am installing version `4.5.2` because we will follow tutorials that have been changed in the latest `4.6.0` version.
+
+```bash
+pip install reframe-hpc==4.5.2
+```
+
+Alternatively, you can
+
+```bash
+git clone -q --depth 1 --branch v4.5.2 https://github.com/reframe-hpc/reframe.git
+source reframe/bootstrap.sh
+```
+
+---
+
+## Hello world example
+
+[Hello world example](https://reframe-hpc.readthedocs.io/en/v4.5.2/tutorial_basics.html#the-hello-world-test)
+
+----
+
+### Include ReFrame modules
+
+The first thing you need is include a few modules from ReFrame. These should be available if the installation step was successful.
+
+```python
+import reframe as rfm
+import reframe.utility.sanity as sn
+```
+
+----
+
+### Create a Test Class
+
+- ReFrame uses decorators to mark classes as tests. 
+- This marks `class HelloTest` as a `rfm.simple_test`.
+- ReFrame tests ultimately derive from `RegressionTest`. There are others derived classes such as `RunOnlyRegressionTest`, we get to those later.
+
+```python
+
+@rfm.simple_test
+class HelloTest(rfm.RegressionTest):
+```
+
+- The data members and methods detailed in the following sections should be placed inside this class.
+
+----
+
+### Add mandatory attributes
+
+- `valid_systems` for where this test can run
+- `valid_prog_environs` for what compilers this test can build with. More on it later.
+- `sourcepath` for source file in a single source test. More on build systems later.
+- Could add `sourcesdir` but it defaults to `src/`
+
+```python
+    valid_systems = ['*']
+    valid_prog_environs = ['*']
+    sourcepath = 'hello.c'
+```
+
+----
+
+### Add sanity check
+
+- ReFrame, by default, makes no assumption about whether a test is successful or not.
+- A test must provide a validation function
+- ReFrame provides a rich set of utility functions that help matching patterns and extract values from the test’s output
+- Here we match a string from stdout
+
+```python
+    @sanity_function
+    def assert_hello(self):
+        return sn.assert_found(r'Hello, World\!', self.stdout)
+```
+
+----
+
+## Builting programming environment
+
+- `reframe --show-config`
+- Builtin programming environment uses `cc` to compile
+
+---
```

Comment:
When are you going to show how to run this benchmark?

Reply:
Now!